vlemish/3SumSolution.cs

```diff
@@ -0,0 +1,80 @@
+﻿namespace _3Sum
+{
+    public class _3SumSolution
+    {
+        // The easiest way to solve the problem is to decrease problem to 2Sum problem
+        // we can do it by creating a dictionary with elements for which we're going to search further.
+        public IList<IList<int>> FindTriples(int[] nums)
+        {
+            // if length is less than 3 -> no triples can be found
+            if (nums.Length < 3)
+            {
+                return new List<IList<int>>();
+            }
+
+            Array.Sort(nums);
+
+            List<Tripple> tripples = new List<Tripple>();
+
+            // create dictionary with elements to search
+            Dictionary<int, int> numsDictionary = new();
+
+            // use pointers to decrease number of iterations
+            // take the first element and add it to dictionary and then use this element to find couple of elements which will give us 0
+            for (int i = 1; i < nums.Length - 1; i++)
+            {
+                var k = i - 1;
+                if (!numsDictionary.ContainsKey(nums[k]))
+                {
+                    numsDictionary.Add(nums[k], k);
+                }
+
+                for (int j = i + 1; j < nums.Length; j++)
+                {
+                    // get a sum of addition i + j and change sign, so we can find an element that will give us 0 as a result
+                    int sum = (nums[i] + nums[j]) * -1;
```

Comment:
I was thinking along those lines as well and for some reason rejected this idea having something like [3, 5, -8]. I thought okay it's either 0 or two numbers of the opposite sign e.g. [0, -5, 5] or [0, 0, 0] and that obviously did not work, but shortening it to sum of two was very bright idea. Nice!